tracecat/executor/service.py

```diff
@@ -86,24 +86,23 @@ async def coro():
 
 
 async def _run_action_direct(
-    *, action: BoundRegistryAction[ArgsClsT], args: ArgsT, validate: bool = False
+    *, action: BoundRegistryAction[ArgsClsT], args: ArgsT
 ) -> Any:
     """Execute the UDF directly.
 
     At this point, the UDF cannot be a template.
     """
-    if validate:
-        # Optional, as we already validate in the caller
-        args = action.validate_args(**args)
     if action.is_template:
-        # This should not be reached
+        # Defensive check
         raise ValueError("Templates cannot be executed directly")
+
+    validated_args = action.validate_args(**args)
```

Comment:
if an exception (RegistryValidationError) is raised, does it get shown nicely in the UI?

Reply:
Unsure

Reply:
Done.